### PR TITLE
chore: adds core organisation tables

### PR DESCRIPTION
**Pull request by jrabbott**
### Context
Adds tables to core database for;

1. Schools
2. Trusts
3. Trust history
4. Financial plans

NB: Schools doesn't have address/location fields at the moment, as there discussion requiring what is actually required. Financial plans is a copy of the existing table

### CODE REVIEW

File: core-infrastructure/src/db/Core.Database/Scripts/006-CreateFinancialPlanTable.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            CreatedBy      nvarchar(255)  NOT NULL,
+            UpdatedAt      datetimeoffset NOT NULL,
+            UpdatedBy      nvarchar(255)  NOT NULL,
+            IsComplete     bit            NOT NULL,
+            Version        int            NOT NULL,
+
+
+            CONSTRAINT PK_FinancialPlan PRIMARY KEY (UKPRN, Year),
+        );
+    END
```
Comment (Chris-Lorch) [inline core-infrastructure/src/db/Core.Database/Scripts/006-CreateFinancialPlanTable.sql:21]: I don't believe we have this one defined in the data models sheet?

File: core-infrastructure/src/db/Core.Database/Scripts/007-CreateTrustTable.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
+IF NOT EXISTS(SELECT *
+              FROM INFORMATION_SCHEMA.TABLES
+              WHERE table_name = 'Trust')
+    BEGIN
+        CREATE TABLE dbo.Trust
+        (
+            UKPRN         nvarchar(50)  NOT NULL,
```
Comment (Chris-Lorch) [inline core-infrastructure/src/db/Core.Database/Scripts/007-CreateTrustTable.sql:7]: Should we distinguish the Trust UKPRN from the UKPRN?

File: core-infrastructure/src/db/Core.Database/Scripts/006-CreateFinancialPlanTable.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
+IF NOT EXISTS(SELECT *
+              FROM INFORMATION_SCHEMA.TABLES
+              WHERE table_name = 'FinancialPlan')
+    BEGIN
+        CREATE TABLE dbo.FinancialPlan
+        (
+            UKPRN          nvarchar(50)   NOT NULL,
```
Comment (Chris-Lorch) [inline core-infrastructure/src/db/Core.Database/Scripts/006-CreateFinancialPlanTable.sql:7]: Would it be better to define the UKPRNs as integers

File: core-infrastructure/src/db/Core.Database/Scripts/006-CreateFinancialPlanTable.sql (hunk 1 above)
Comment (jrabbott) [reply core-infrastructure/src/db/Core.Database/Scripts/006-CreateFinancialPlanTable.sql:21]: It's the existing table for Financial plans. No impact on the data pipeline.

File: core-infrastructure/src/db/Core.Database/Scripts/006-CreateFinancialPlanTable.sql (hunk 3 above)
Comment (jrabbott) [reply core-infrastructure/src/db/Core.Database/Scripts/006-CreateFinancialPlanTable.sql:7]: Keeping it flexible for the moment, as we discussion be might be to change this in future.

File: core-infrastructure/src/db/Core.Database/Scripts/008-CreateSchoolTable.sql [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            HeadteacherEmail    nvarchar(255) NOT NULL,
+
+            CONSTRAINT PK_School PRIMARY KEY (UKPRN),
+        );
+
+        CREATE INDEX School_URN ON School (URN)
+        CREATE INDEX School_TrustUKPRN ON School (TrustUKPRN)
+        CREATE INDEX School_FederationLeadUKPRN ON School (FederationLeadUKPRN)
+        CREATE INDEX School_LACode ON School (LACode)
+    END
```
Comment (Chris-Lorch) [inline core-infrastructure/src/db/Core.Database/Scripts/008-CreateSchoolTable.sql:39]: I think there are a few items missing which were defined in the data model spreadsheet. What are we doing about addresses and London weightings here?

File: core-infrastructure/src/db/Core.Database/Scripts/007-CreateTrustTable.sql (hunk 2 above)
Comment (jrabbott) [reply core-infrastructure/src/db/Core.Database/Scripts/007-CreateTrustTable.sql:7]: It's the trust table, so we don't need it.

File: core-infrastructure/src/db/Core.Database/Scripts/008-CreateSchoolTable.sql (hunk 4 above)
Comment (jrabbott) [reply core-infrastructure/src/db/Core.Database/Scripts/008-CreateSchoolTable.sql:39]: There's a note in the description - left out the address for the moment

**Review by Chris-Lorch — APPROVED** [review]

File: core-infrastructure/src/db/Core.Database/Scripts/008-CreateSchoolTable.sql (hunk 4 above)
Comment (jrabbott) [reply core-infrastructure/src/db/Core.Database/Scripts/008-CreateSchoolTable.sql:39]: London weighting is there.

**Pull request merged**